### PR TITLE
executor: make auto analyze more conservative

### PR DESCRIPTION
**Pull request by lamxTyler**
PTAL @shenli  @hanfei1991  @winoros

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -209,6 +210,10 @@ func (a *statement) buildExecutor(ctx context.Context) (Executor, error) {
 	}
 	if _, ok := a.plan.(*plan.Analyze); ok && ctx.GetSessionVars().InRestrictedSQL {
 		priority = kv.PriorityLow
+		ctx.GetSessionVars().Systems[variable.TiDBBuildStatsConcurrency] = "1"
+		ctx.GetSessionVars().DistSQLScanConcurrency = 1
+		ctx.GetSessionVars().IndexSerialScanConcurrency = 1
+		ctx.GetSessionVars().Systems[variable.TxnIsolation] = ast.ReadCommitted
```
Comment (shenli) [inline executor/adapter.go]: Restore the session context after running analyze.

**Comment by shenli** [comment]
We should investigate why analyze use so many memory.

**Comment by lamxTyler** [comment]
@shenli  One of the reason is that analyze will open all the sub-executors at begin, but it is not necessary, we need only open those executors when build histograms.

**Comment by shenli** [comment]
LGTM
@hanfei1991 PTAL

**Pull request merged**